worlds/mmx/__init__.py

```diff
@@ -0,0 +1,402 @@
+
+        # Add sub tanks into the pool
+        if "Sub Tanks" in sigma_open and self.options.sigma_sub_tank_count.value > 0:
+            i = self.options.sigma_sub_tank_count.value
+            itempool += [self.create_item(ItemName.sub_tank) for _ in range(i)]
+            if i != 4:
+                itempool += [self.create_item(ItemName.sub_tank, ItemClassification.useful) for _ in range(4 - i)]
+        else:
+            itempool += [self.create_item(ItemName.sub_tank, ItemClassification.useful) for _ in range(4)]
+
+        # Add junk items into the pool
+        junk_count = total_required_locations - len(itempool)
+
+        junk_weights = []
+        junk_weights += ([ItemName.small_hp] * 30)
+        junk_weights += ([ItemName.large_hp] * 40)
+        junk_weights += ([ItemName.life] * 30)
+
+        junk_pool = []
+        for i in range(junk_count):
+            junk_item = self.random.choice(junk_weights)
+            junk_pool.append(self.create_item(junk_item))
+
+        itempool += junk_pool
+
+        # Set Maverick Medals
+        maverick_location_names = [
+            LocationName.armored_armadillo_clear,
+            LocationName.boomer_kuwanger_clear,
+            LocationName.chill_penguin_clear,
+            LocationName.flame_mammoth_clear,
+            LocationName.launch_octopus_clear,
+            LocationName.spark_mandrill_clear,
+            LocationName.sting_chameleon_clear,
+            LocationName.storm_eagle_clear
+        ]
+        for location_name in maverick_location_names:
+            self.multiworld.get_location(location_name, self.player).place_locked_item(self.create_item(ItemName.maverick_medal))
+
+        # Set victory item
+        self.multiworld.get_location(LocationName.sigma_fortress_4_sigma, self.player).place_locked_item(self.create_item(ItemName.victory))
+
+        # Finish
+        self.multiworld.itempool += itempool
+
+
+    def create_item(self, name: str, force_classification: ItemClassification = False) -> Item:
+        data = item_table[name]
+
+        if force_classification:
+            classification = force_classification
+        elif data.progression:
+            classification = ItemClassification.progression
+        elif data.trap:
+            classification = ItemClassification.trap
+        else:
+            classification = ItemClassification.filler
+        
+        created_item = MMXItem(name, classification, data.code, self.player)
+
+        return created_item
+
+
+    def set_rules(self) -> None:
+        if hasattr(self.multiworld, "generation_is_fake"):
+            if hasattr(self.multiworld, "re_gen_passthrough"):
+                if "Mega Man X" in self.multiworld.re_gen_passthrough:
+                    self.boss_weaknesses = self.multiworld.re_gen_passthrough["Mega Man X"]["weakness_rules"]
+        build_rules(self)
+
+
+    def fill_slot_data(self) -> Dict[int, Any]:
```

Comment:
```suggestion
    def fill_slot_data(self) -> Dict[str, Any]:
```
I wonder how that got there...

Reply:
oops xd
I'm too new to typing and all that stuff.